logging/src/main/java/org/jboss/as/logging/LoggingResource.java

```diff
@@ -126,7 +125,19 @@ public Resource requireChild(final PathElement element) {
     @Override
     public boolean hasChildren(final String childType) {
         if (LOG_FILE.equals(childType)) {
-            return !getChildrenNames(LOG_FILE).isEmpty();
+            final String logDir = pathManager.getPathEntry(ServerEnvironment.SERVER_LOG_DIR).resolvePath();
+            if (logDir != null) {
+                final Path dir = Paths.get(logDir);
+                final Collection<String> validFileNames = findValidFileNames(getFileHandlersModel());
+                // If there's at least one child, then we have children
+                for (String name : validFileNames) {
+                    final Path file = dir.resolve(name);
+                    if (Files.exists(file) && Files.isReadable(file)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
```

Comment:
I'm not particularly arguing for it or against it. It's definitely an edge case that someone would delete the base file but then still expect to be able to use the management API to read the other ones.

Perhaps you can optimize the normal case (which you have here) and also optimize the "no files in the dir at all" case, and then have the slow approach for the hopefully unusual corner case. Or something like that.